### PR TITLE
[RFC] CONTRIBUTING.md: describe adoption/orphaning policy and procedure

### PR DESCRIPTION
**Pull request by classabbyamp**
I think it would be good to codify some of this. Open to suggestions on everything.

@void-linux/pkg-committers

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,6 +82,25 @@ The checksum can be updated automatically with the `xgensum` helper from the [xt
...
+
+It is best to adopt a template when making another change to it. When adopting the template, add your name or username
+and email to the `maintainer` field in the template, and mention the adoption in your commit message, for example:
+
+    libfoo: update to 1.2.3, adopt.
+
+### Orphaning a template
+
+If you no longer wish to maintain a template, you can remove yourself as maintainer by setting the `maintainer` field in
+the template to `Orphaned <orphan@voidlinux.org>`. The commit message should mention this, for example:
```
Comment (mhmdanas) [inline CONTRIBUTING.md:100]: Might also want to mention that a package can be orphaned without having to make another change to the template?

**Comment by icp1994** [comment]
`package` instead of `template` sounds more natural to me.

**Comment by classabbyamp** [comment]
template is used throughout most of the rest of the document

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -82,6 +82,25 @@ The checksum can be updated automatically with the `xgensum` helper from the [xt
 
     $ xgensum -i <pkgname>
 
+### Adopting a template
+
+If a template is orphaned (maintained by `orphan@voidlinux.org`) or the current `maintainer` has not contributed in a long
+time (over a year), template maintainership can be adopted by someone else. To ensure a template gets the care it needs,
+template adopters should use the package in question and have a history of contributing changes to that template (updating
+the template several times, for example).
```
Comment (ahesford) [inline CONTRIBUTING.md:90]:
```suggestion
If a template is orphaned (maintained by `orphan@voidlinux.org`) or the current `maintainer` has not made contributions to Void in at least a year, template maintainership can be adopted by someone else. To ensure a template gets the care it needs, users wishing to adopt a template should be familiar with the package and have an established history of contributions to Void. Those who have contributed several updates, especially for the template in question, are good candidates for adoption.
```

Comment (mhmdanas) [reply CONTRIBUTING.md:90]: My take:
```suggestion
If a template is orphaned (maintained by `orphan@voidlinux.org`) or the current maintainer has not made contributions to Void in at least a year, the template can be orphaned or adopted by someone else. To ensure a template gets the care it needs, users wishing to adopt a template should be familiar with the package and have an established history of contributions to Void. Those who have contributed several updates, especially to the template in question, are good candidates for adoption.
```

**Comment by Vaelatern** [comment]
Please remember all we are not codifying package adoption and orphaning in code. We're trying to codify the process in a little way so we don't see people adopt a package in their first ever contribution, taking over from an active maintainer.

**Comment by classabbyamp** [comment]
feedback incorporated

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -82,6 +82,29 @@ The checksum can be updated automatically with the `xgensum` helper from the [xt
 
     $ xgensum -i <pkgname>
 
+### Adopting a template
+
+If a template is orphaned (maintained by `orphan@voidlinux.org`) or the current `maintainer` has not contributed to
```
Comment (mhmdanas) [inline CONTRIBUTING.md:87]:
```suggestion
If a template is orphaned (maintained by `orphan@voidlinux.org`) or the current maintainer has not contributed to
```

Comment (classabbyamp) [reply CONTRIBUTING.md:87]: I code-quoted it to indicate that it's the variable in the template

**Review by Vaelatern — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (mhmdanas) [reply CONTRIBUTING.md:87]: Fine then.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -82,6 +82,29 @@ The checksum can be updated automatically with the `xgensum` helper from the [xt
...
+
+### Orphaning a template
+
+If you no longer wish to maintain a template, you can remove yourself as maintainer by setting the `maintainer` field in
+the template to `Orphaned <orphan@voidlinux.org>`. The commit message should mention this, for example:
+
+    libfoo: orphan.
+
+It is not necessary to make other changes to the template when orphaning, and incrementing the revision (triggering a
+rebuild) is not necessary.
```
Comment (mhmdanas) [inline CONTRIBUTING.md:106]:
```suggestion
rebuild) is not necessary either.
```

After this it's good to go for me.

**Review by Piraty — APPROVED** [review]

**Pull request merged**